src/SequenceManager.cc

```diff
@@ -60,8 +60,14 @@ std::vector<SequenceManager::SequenceOnDisk> SequenceManager::list_sequences() c
         }
         else
         {
-            // No unique ID in folder name? Automatically generate one and rename the
-            // folder.
+            // No unique ID in folder name? It could be a non-sequence folder or a
+            // sequence folder from an older version. We only believe it is the latter if
+            // it contains a sequence.lua file.
+            if (not std::filesystem::exists(entry.path() / "sequence.lua"))
+                continue;
+
+            // So it is a sequence folder after all. We automatically generate a unique ID
+            // and rename the folder.
```

Comment:
I'm going through the MR commit-by-commit.

This whole commit would be unneeded if we forgo automatic backward compatibility on a library level and rather handle the braking change in Sequence store-folder-names on a server level via debian maintainer scripts.

Reply:
To cite myself from above:

> We disagree on how to migrate old sequences to the new format. I am too lazy to fiddle with shell scripts for the Debian packages, especially because the C++ code already solves the problem.

So my preferred solution would be to leave this part of the code as-is, but I am open to further arguments, maybe also from @wyrnat, @MissPeppermintPatty.

Reply:
The thing is that the folder names on disk will change. Before:
```
My_favorite_sequence
Another_one
```
After:
```
My_favorite_sequence [ab26b5107a5f72c]
Another_one [b51e7a5f72c35970]
```
The code in this MR will automatically do the renaming if it finds an old-school sequence folder. @Finii is arguing for letting some script in the Debian package do that instead.